services/get-students.php

```diff
@@ -0,0 +1,89 @@
+<?php
+// File: /services/get_students.php
+
+// Set the content type to JSON
+header('Content-Type: application/json');
+
+// Shared secret for authorization
+$sharedSecret = "gradeplus";
+
+// Check authorization
+if ($authorize !== $sharedSecret) {
+    header("Location: illegal.php");
+    exit();
+}
+
+// Database credentials
+$db_host = 'localhost';
+$db_user = 'gradeplusclient';
+$db_pass = 'gradeplussql';
+$db_name = 'gradeplus';
+
+// Create a new MySQLi connection
+$conn = new mysqli($db_host, $db_user, $db_pass, $db_name);
```

Comment:
Remove single use variables and directly put in values in `$conn`
```suggestion
$conn = new mysqli('localhost','gradeplusclient' , 'gradeplussql', 'gradeplus');
```